### PR TITLE
Homework2: Maxim Sankouski 

### PR DESCRIPTION
**Pull request by MaxSnk**

### CODE REVIEW

File: src/2_2_5.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Input: A positive integer.
+
+Output: The product of the digits as an integer.
+
+Precondition: 0 < number < 10**6"""
+
+
+def checkio(number: int) -> int:
+    if 0 < number < 10 ** 6:
```
Comment (belovmd) [inline src/2_2_5.py:15]: why do we need this limitations?

**Review by belovmd — CHANGES_REQUESTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-python-spring/blob/master/homework_review/Homework2.md):
- 2
- 4
- 6

File: src/2_2_5.py (hunk 1 above)
Comment (MaxSnk) [reply src/2_2_5.py:15]: This is verification of the precondition.

Comment (belovmd) [reply src/2_2_5.py:15]: If you check conditions you should handle if they are out of limits. here it's just redundant if

File: src/2_2_7.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+re.match("[a-zA-Z0-9]+", password)
+0 < len(password) ≤ 64
+"""
+
+
+import re
+
+
+def checkio(data: str) -> bool:
+    if 10 <= len(data) <= 64:
```
Comment (belovmd) [inline src/2_2_7.py:31]: redundant

File: src/2_2_7.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        if re.search('[0-9]', data)\
```
Comment (belovmd) [inline src/2_2_7.py:32]: bad formatting

File: src/2_7.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+number = number_iter = int(input("Insert the number: "))
+
+new_number = 0
+
+number_len = int(math.log10(number) + 1)
+
+for number_len in range(number_len, 0, -1):
+    new_number = new_number * 10 + number_iter % 10
+    number_iter = int(number_iter / 10)
```
Comment (belovmd) [inline src/2_7.py:20]: unclear naming

File: src/2_6_Fibo.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+n = int(input("Insert the number: "))
+
+fibo_n_minus_1 = 0
+fibo_n = 1
+
+i = 1
+
+while i < n:
+    fibo_n += fibo_n_minus_1
+    fibo_n_minus_1 = fibo_n - fibo_n_minus_1
```
Comment (belovmd) [inline src/2_6_Fibo.py:17]: unclear naming

**Review by belovmd — APPROVED** [review]
8
these tasks don't look pretty difficult (4 paragraph of review)

**Pull request closed without merging**